### PR TITLE
[Merged by Bors] - Generate TLS certificates as part of the test

### PR DESCRIPTION
**Pull request by fasmat**
## Motivation
Follow up to #5347: instead of re-using generated templates create them as part of the test.

## Changes
- Update test to generate the certificates it needs

## Test Plan
- existing tests pass

## TODO

- [x] Explain motivation or link existing issue(s)
- [x] Test changes and document test plan
- [x] Update documentation as needed
- [x] Update [changelog](../CHANGELOG.md) as needed

### CODE REVIEW

**Review by lrettig — COMMENTED** [review]
README changes appear unrelated

**Comment by fasmat** [comment]
> README changes appear unrelated

They have been taken from the deleted `README.md` in the `testdata` folder since I didn't want this information to get lost and it will be relevant for smeshers running a remote PoST service. The main `README.md` felt like the right place for it 🙂

File: api/grpcserver/grpcserver_tls_test.go [hunk 1]
```diff
@@ -1,32 +1,175 @@
...
+	clientCertName = "client.crt"
+	clientKeyName  = "client.key"
 )
 
-func launchTLSServer(tb testing.TB, services ...ServiceAPI) (Config, func()) {
-	pwd, err := os.Getwd()
+func genKeys(tb testing.TB) string {
+	dir := tb.TempDir() // to store the generated files
+
+	caKey, err := rsa.GenerateKey(rand.Reader, 4096)
```
Comment (poszu) [inline api/grpcserver/grpcserver_tls_test.go:32]: Would it make sense to factor this part out?

Edit: I mean the repeated block that creates the thing

**Review by poszu — APPROVED** [review]

File: api/grpcserver/grpcserver_tls_test.go (hunk 1 above)
Comment (fasmat) [reply api/grpcserver/grpcserver_tls_test.go:32]: The problem is that the code block creating the keys and certificates is always slightly different, I'll see what I can do.

**Comment by fasmat** [comment]
bors merge

**Pull request closed without merging**